tasks/configure.yml

```diff
@@ -109,26 +103,10 @@
     group: root
     mode: 0644
   loop: "{{ _mariadb_systemd_override }}"
-  register: _mariadb_systemd_conf
-
-- name: Reload systemd on config changes right away
-  ansible.builtin.systemd:
-    daemon_reload: true
-  when: _mariadb_systemd_conf is changed
-  tags: [skip_ansible_lint]
-
-- name: "Restart MariaDB on config changes right away"
-  ansible.builtin.service:
-    name: "{{ _mariadb_daemon }}"
-    state: restarted
-  when: _mariadb_main_config is changed or _mariadb_additional_config is changed
-  tags: [skip_ansible_lint]
+  notify: Restart MariaDB on config changes
 
-- name: "Set startup settings and ensure MariaDB is started"
-  ansible.builtin.service:
-    name: "{{ _mariadb_daemon }}"
-    state: started
-    enabled: "{{ mariadb_enabled_on_startup | d(true) }}"
+- name: "Flush handlers"
```

Comment:
Assuming this role is used as part of a playbook with other tasks, roles that are also using handlers.

Flushing handlers here would run external handlers here too. If the other tasks,roles in the playbook relied on the fact that Ansible executes the handlers at the end of a playbook run, this could cause side effects inside the playbook.

I think this behaviour should be mentionned in the readme.

Reply:
Flush_handlers inside a role, will only flush handers that have been defined within the role.

No impact for all the playbook.